gobblin-runtime/src/main/java/gobblin/runtime/FsDatasetStateStore.java

```diff
@@ -254,16 +246,47 @@ public void persistDatasetState(String datasetUrn, JobState.DatasetState dataset
     String jobName = datasetState.getJobName();
     String jobId = datasetState.getJobId();
 
-    datasetUrn = CharMatcher.is(':').replaceFrom(datasetUrn, '.');
-    String tableName = Strings.isNullOrEmpty(datasetUrn) ? jobId + DATASET_STATE_STORE_TABLE_SUFFIX
-        : datasetUrn + "-" + jobId + DATASET_STATE_STORE_TABLE_SUFFIX;
+    Optional<String> sanitizedDatasetUrn = DatasetUrnSanitizer.sanitize(datasetUrn);
+    String tableName = sanitizedDatasetUrn.isPresent() ?
+        sanitizedDatasetUrn.get() + StateStoreTableInfo.TABLE_PREFIX_SEPARATOR + jobId : jobId;
     LOGGER.info("Persisting " + tableName + " to the job state store");
     put(jobName, tableName, datasetState);
-    createAlias(jobName, tableName, getAliasName(datasetUrn));
```

Comment:
`current.jst` will not be written to or read from after this change. States are always written to new files and when `current.jst` is requested from the state store, the latest file will be found and returned.